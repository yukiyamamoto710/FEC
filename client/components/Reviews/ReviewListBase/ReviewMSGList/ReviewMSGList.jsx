import React from 'react';
import PropTypes from 'prop-types';
import Message from '../../Message/Message';
import Photos from '../../Photos/Photos';
import MSGHeader from './MSGHeader/MSGHeader';
import HelpfulMSG from '../../HelpfulMSG/HelpfulMSG';

const MessageList = (props) => {
  const {
    listReviews,
    reported,
  } = props;
  return (
    <div
      data-testid="msgListBase"
      className="msgListBase"
    >
      { listReviews.map((i, index) => (
        <div
          key={index}
          className="msgContainer"
        >
          <MSGHeader
            user={i.reviewer_name}
            date={i.date}
            rate={i.rating}
          />
          <div className="msgSummary ">
            {i.summary}
          </div>
          <Message body={i.body} />
          {i.recommend
            ? (
              <div className="msgBody">
                âœ“ I recommend this product
              </div>
            )
            : null}
          {(i.response)
            ? (
              <div className="msgResponseContainer">
                Response:
                <br />
                <div className="msgResponse">
                  {i.response}
                </div>
              </div>
            )
            : null}
          <Photos
            photos={i.photos}
          />
          <HelpfulMSG
            countHelpful={i.helpfulness}
            reviewId={i.review_id}
            countNotHelpful={0}// should get the key from i
            reported={reported}
          />
        </div>
      ))}
    </div>
  );
};

MessageList.propTypes = {
  reported: PropTypes.func,
  listReviews: PropTypes.arrayOf(PropTypes.shape({})),
};

MessageList.defaultProps = {
  reported: () => 1,
  listReviews: [{}],
};
export default MessageList;
