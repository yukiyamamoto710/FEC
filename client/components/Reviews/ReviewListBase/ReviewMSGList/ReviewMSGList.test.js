/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import ReviewMSGList from './ReviewMSGList';

afterEach(cleanup);

it('should not render text', () => {
  render(
    <ReviewMSGList
      listReviews={[{}]}
      reported={jest.fn()}
    />,
  );
  const test = screen.queryByText('✓ I recommend this product');
  const test1 = screen.queryByText('Response:');
  expect(test).not.toBeInTheDocument();
  expect(test1).not.toBeInTheDocument();
});

it('should render text', () => {
  render(
    <ReviewMSGList
      listReviews={[{ response: 'hi', recommend: true }]}
      reported={jest.fn()}
    />,
  );
  const test = screen.queryByText('✓ I recommend this product');
  const test1 = screen.queryByText('Response:');
  expect(test).toBeInTheDocument();
  expect(test1).toBeInTheDocument();
});

it('should have default reported', () => {
  expect(ReviewMSGList.defaultProps.reported()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <ReviewMSGList
      listReviews={[{ response: 'hi', recommend: true }]}
      reported={jest.fn()}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
