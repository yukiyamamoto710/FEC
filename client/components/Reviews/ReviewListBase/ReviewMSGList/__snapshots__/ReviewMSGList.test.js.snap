// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="msgListBase"
  data-testid="msgListBase"
>
  <div
    className="msgContainer"
  >
    <div
      className="msgHeader"
      data-testid="msgHeader"
    >
      <div
        className="ratingHead"
        data-testid="stars"
      >
        <div>
          <div
            className="msgStar emptyStar"
            data-testid="button0"
            id=" 1"
            label="emptyStar"
            name=""
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
        <div>
          <div
            className="msgStar emptyStar"
            data-testid="button1"
            id=" 2"
            label="emptyStar"
            name=""
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
        <div>
          <div
            className="msgStar emptyStar"
            data-testid="button2"
            id=" 3"
            label="emptyStar"
            name=""
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
        <div>
          <div
            className="msgStar emptyStar"
            data-testid="button3"
            id=" 4"
            label="emptyStar"
            name=""
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
        <div>
          <div
            className="msgStar emptyStar"
            data-testid="button4"
            id=" 5"
            label="emptyStar"
            name=""
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
      </div>
      <div>
        
        ,
         
        lid Date
      </div>
    </div>
    <div
      className="msgSummary "
    />
    <div
      className="msgBody"
      data-testid="textHidden"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      
    </div>
    <div
      className="msgBody"
    >
      âœ“ I recommend this product
    </div>
    <div
      className="msgResponseContainer"
    >
      Response:
      <br />
      <div
        className="msgResponse"
      >
        hi
      </div>
    </div>
    <div
      className="photoContainer"
      data-testid="photos"
    >
      <div
        data-testid="img"
      >
        <img
          alt=""
          className="photos"
        />
      </div>
    </div>
    <div>
      <button
        className="helpfulButton"
        data-testid="helpfulButton"
        onClick={[Function]}
        type="button"
      >
        Helpful? Yes (0) 
      </button>
      <button
        className="helpfulButton"
        data-testid="nothelpfulButton"
        onClick={[Function]}
        type="button"
      >
         No (0) 
      </button>
      <button
        className="helpfulButton"
        data-testid="report"
        onClick={[Function]}
        type="button"
      >
         | report
      </button>
    </div>
  </div>
</div>
`;
